analysis.go

```diff
@@ -161,92 +100,80 @@ func checkProjectTask(checkFactory getProjectCheckFactory, JSONResourceFactory D
-func CheckEventLogForErrors(JSONResourceFactory DumpedJSONResourceFactory) (Result, error) {
-	result := Result{Status: analysisErrorNotDiscovered, StatusMessage: "this issue was not detected", CheckName: "check eventlog for any errors", Info: []Info{}, Events: []Event{}}
-	events := Events{}
-	if err := JSONResourceFactory(filepath.Join("definitions", "events.json"), &events); err != nil {
-		result.Status = analysisErrorReadingDumpedResource
-		result.StatusMessage = "Error executing task: " + err.Error()
-		return result, err
+func CheckEventLogForErrors(events types.EventList) Result {
+	result := Result{
+		CheckName:     "check eventlog for any errors",
+		Status:        analysisErrorNotDiscovered,
+		StatusMessage: "this issue was not detected",
 	}
-
 	for _, event := range events.Items {
 		if event.Type != "Normal" {
 			result.Status = analysisErrorDiscoveredByAnalysis
 			result.StatusMessage = "Errors detected in event log"
 			result.Events = append(result.Events, event)
 		}
 	}
-
-	return result, nil
+	return result
 }
 
 // CheckDeployConfigsReplicasNotZero checks all deployment configs in the
 // supplied JSON Resource Factory, and if any are found with a replica of 0, it
 // will add a note about it to the returned result.
-func CheckDeployConfigsReplicasNotZero(ResourceFactory DumpedJSONResourceFactory) (Result, error) {
-	result := Result{Status: analysisErrorNotDiscovered, StatusMessage: "this issue was not detected", CheckName: "check deployconfig replicas not 0", Info: []Info{}, Events: []Event{}}
-	deploymentConfigs := DeploymentConfigs{}
-	err := ResourceFactory(filepath.Join("definitions", "deploymentconfigs.json"), &deploymentConfigs)
-	if err != nil {
-		result.Status = analysisErrorReadingDumpedResource
-		result.StatusMessage = "Error executing task: " + err.Error()
-		return result, err
+func CheckDeployConfigsReplicasNotZero(deploymentConfigs types.DeploymentConfigList) Result {
+	result := Result{
+		CheckName:     "check deployconfig replicas not 0",
+		Status:        analysisErrorNotDiscovered,
+		StatusMessage: "this issue was not detected",
 	}
-
 	for _, deploymentConfig := range deploymentConfigs.Items {
 		if deploymentConfig.Spec.Replicas == 0 {
-			info := Info{Name: deploymentConfig.Metadata.Name, Namespace: deploymentConfig.Metadata.Namespace, Kind: deploymentConfig.Kind, Count: 1, Message: "the replica parameter is set to 0, this should be greater than 0"}
 			result.Status = analysisErrorDiscoveredByAnalysis
 			result.StatusMessage = "one or more deployConfig replicas are set to 0"
-			result.Info = append(result.Info, info)
+			result.Info = append(result.Info, Info{
+				Name:      deploymentConfig.ObjectMeta.Name,
+				Namespace: deploymentConfig.ObjectMeta.Namespace,
+				Kind:      deploymentConfig.Kind, // REVIEW: this gets mixed information -- DeploymentConfig, Event, ..., what's the purpose?
+				Count:     1,                     // REVIEW: is Count ever going to be != 1?
```

Comment:
@philbrookes one more question, what is the idea behind `Count`?
Seems that we don't increment it when we find another issue.